### PR TITLE
Fixes #13071: Help option for commands was not working in modules

### LINKED ISSUE #13071: Help option does not work in sub-module command
Since 2.0.10, Yii2 add a help option for command controllers by default, but when the command is in submodule, that dose not work.

Fixing this is very simple, just modify [yii\console\Controller#L125](https://github.com/yiisoft/yii2/blob/master/framework/console/Controller.php#L125), 
change `$this->id` to `$this->uniqueId`.

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13071

### CODE REVIEW

**Comment by arogachev** [comment]
@dynasource There were no tests for #12028. Added unit tests to cover both #12028 and #13079 issues.

File: tests/framework/console/ControllerTest.php [hunk 1]
```diff
@@ -89,4 +94,54 @@ public function testResponse()
...
+        $controller = new FakeController('posts', new Module('news'));
+        $controller->help = true;
+        $controller->runAction('index');
+
+        $this->assertFalse(FakeController::getWasActionIndexCalled());
+        $this->assertEquals(FakeHelpController::getActionIndexLastCallParams(), ['news/posts/index']);
+    }
+}
+
+class FakeHelpController extends HelpController
```
Comment (samdark) [inline tests/framework/console/ControllerTest.php]: It's better to extract this class into separate file...

**Review by samdark — CHANGES_REQUESTED** [review]
The change itself is correct. I'd like to ask to extract 2nd class in the test to its own file. Otherwise, it's good.

File: tests/framework/console/ControllerTest.php (hunk 1 above)
Comment (arogachev) [inline tests/framework/console/ControllerTest.php]: OK. I'll do this tomorrow.

**Review by cebe — APPROVED** [review]

File: tests/framework/console/ControllerTest.php [hunk 2]
```diff
@@ -89,4 +94,54 @@ public function testResponse()
...
+            'help' => 'yiiunit\framework\console\FakeHelpController',
+        ];
+
+        $controller = new FakeController('posts', Yii::$app);
+        $controller->runAction('index');
+
+        $this->assertTrue(FakeController::getWasActionIndexCalled());
+        $this->assertNull(FakeHelpController::getActionIndexLastCallParams());
+
+        $controller = new FakeController('posts', Yii::$app);
```
Comment (dynasource) [inline tests/framework/console/ControllerTest.php]: here you are starting a new 'test' setting. To have it perfectly self contained, you should add this to a separate test method. The test will also easier to understand.

File: tests/framework/console/ControllerTest.php [hunk 3]
```diff
@@ -89,4 +94,54 @@ public function testResponse()
...
+        $controller->help = true;
+        $controller->runAction('index');
+
+        $this->assertFalse(FakeController::getWasActionIndexCalled());
+        $this->assertEquals(FakeHelpController::getActionIndexLastCallParams(), ['posts/index']);
+
+        // @see https://github.com/yiisoft/yii2/issues/13071
+
+        $controller = new FakeController('posts', new Module('news'));
```
Comment (dynasource) [inline tests/framework/console/ControllerTest.php]: same here. New test method with a nice name stating what is being tested.

**Review by dynasource — CHANGES_REQUESTED** [review]
thumbs up for the unit tests!

**Review by dynasource — APPROVED** [review]
well done

**Review by cebe — APPROVED** [review]

File: tests/framework/console/ControllerTest.php (hunk 2 above)
Comment (dynasource) [inline tests/framework/console/ControllerTest.php]: 'crucial'....it depends on which hat you put on. From the consistency & readability & stability perspective, it is a better practice.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**